src/Disks/DiskCacheWrapper.cpp

```diff
@@ -217,6 +217,9 @@ void DiskCacheWrapper::moveFile(const String & from_path, const String & to_path
 {
     if (cache_disk->exists(from_path))
     {
+        if (cache_disk->exists(to_path) && cache_disk->isDirectory(to_path))
```

Comment:
It's unclear for me: looks like we are clearing the contents of the destination directory. Does not like that this is how it supposed to work.

Reply:
There can be a situation that you move the cache directory and then CH process is terminated for some reasons.
In this case, after you try to move it again, you will get an exception that the directory already exists.
Such error is demonstrated in PR description.

Reply:
It's not described in comments in code.

Reply:
But I don't understand why the `moveFile` method is used to move directories?

Reply:
I think because `moveFile` and `moveDirectory` had no difference in their effect (same implementation).
I've changed `moveFile` invocations to `moveDirectory` where it's to be supposed in code and also moved directory cleanup fix to `moveDirectory` method.